moveit_experimental/moveit_jog_arm/include/moveit_jog_arm/jog_arm_data.h

```diff
@@ -38,34 +38,53 @@
 
 #pragma once
 
-// System
-#include <mutex>
-#include <thread>
+// Boost
+#include <boost/lockfree/spsc_queue.hpp>
 
 // Eigen
 #include <Eigen/Geometry>
 
 // ROS
 #include <control_msgs/JointJog.h>
 #include <geometry_msgs/TwistStamped.h>
-#include <sensor_msgs/JointState.h>
 #include <trajectory_msgs/JointTrajectory.h>
 
 // moveit_jog_arm
 #include "status_codes.h"
 
 namespace moveit_jog_arm
 {
-// Variables to share between threads, and their mutexes
-// Inherit from a mutex so the struct can be locked/unlocked easily
-struct JogArmShared : public std::mutex
+// Types for queues between threads
+constexpr size_t QUEUE_SIZE = 10;
+template <typename T, size_t U>
+using MessageQueue = boost::lockfree::spsc_queue<T, boost::lockfree::capacity<U>>;
+using TwistedStampedQueue = MessageQueue<geometry_msgs::TwistStamped, QUEUE_SIZE>;
+using JointJogQueue = MessageQueue<control_msgs::JointJog, QUEUE_SIZE>;
+using JointTrajectoryQueue = MessageQueue<trajectory_msgs::JointTrajectory, QUEUE_SIZE>;
+
+// Helper function for poping the latest value out of a queue
```

Comment:
poping --> popping

Unless you mean this:  https://media.irishpost.co.uk/uploads/2018/06/11150144/pope-francis-swag2.jpg

Reply:
lol, fixed